### PR TITLE
refactor: admin journeys provider and useAdminJourneyQuery hook

### PR DESCRIPTION
**Pull request by storyworks**
# Description

### <samp>🤖 Generated by Copilot at cdb7c8b</samp>

This pull request updates the journeys-admin app to fetch and display more information about the journeys, such as user email, team members, tags, and published status. It also refactors the data fetching logic to use a custom hook that provides more functionality and control than the generic one. Additionally, it renames the `PublisherTemplate` type to `Journey` in the GraphQL schema and updates the corresponding fields and values in various components and queries.

- Link to Basecamp Todo

# How should this PR be QA Tested?

Please describe the QA tests that you ran to verify your changes. Please also note any relevant details for your test configuration.

- [ ] Test A
- [ ] Test B

# Walkthrough

### <samp>🤖 Generated by Copilot at cdb7c8b</samp>

*  Import `UserTeamRole` type and add `email`, `team`, and `tags` fields to `GetAdminJourney` and `GetJourney` queries ([link](https://github.com/JesusFilm/core/pull/1940/files?diff=unified&w=0#diff-9fecf255f6a72895588618cbc843b8a568d1f488e250d433135d9762c70dc518L6-R6), [link](https://github.com/JesusFilm/core/pull/1940/files?diff=unified&w=0#diff-9fecf255f6a72895588618cbc843b8a568d1f488e250d433135d9762c70dc518R598), [link](https://github.com/JesusFilm/core/pull/1940/files?diff=unified&w=0#diff-9fecf255f6a72895588618cbc843b8a568d1f488e250d433135d9762c70dc518L628-R670), [link](https://github.com/JesusFilm/core/pull/1940/files?diff=unified&w=0#diff-9fecf255f6a72895588618cbc843b8a568d1f488e250d433135d9762c70dc518R694), [link](https://github.com/JesusFilm/core/pull/1940/files?diff=unified&w=0#diff-1ec4c02e97a38e8aed295fe74ec2c3ac54754837958d07baf3587e7f8e9cf427L6-R6), [link](https://github.com/JesusFilm/core/pull/1940/files?diff=unified&w=0#diff-1ec4c02e97a38e8aed295fe74ec2c3ac54754837958d07baf3587e7f8e9cf427R598), [link](https://github.com/JesusFilm/core/pull/1940/files?diff=unified&w=0#diff-1ec4c02e97a38e8aed295fe74ec2c3ac54754837958d07baf3587e7f8e9cf427L628-R670), [link](https://github.com/JesusFilm/core/pull/1940/files?diff=unified&w=0#diff-1ec4c02e97a38e8aed295fe74ec2c3ac54754837958d07baf3587e7f8e9cf427R694))
* Replace `useQuery` with `useAdminJourneyQuery` custom hook and remove unused imports in journey pages (`apps/journeys-admin/pages/journeys/[journeyId].tsx`, `apps/journeys-admin/pages/journeys/[journeyId]/reports.tsx`, `apps/journeys-admin/pages/journeys/[journeyId]/reports/visitors.tsx`) ([link](https://github.com/JesusFilm/core/pull/1940/files?diff=unified&w=0#diff-3873b898541aff5b78e0bc4aba59192c3d0e767d86c2e57185f5c88dd3f02063L1-R1), [link](https://github.com/JesusFilm/core/pull/1940/files?diff=unified&w=0#diff-3873b898541aff5b78e0bc4aba59192c3d0e767d86c2e57185f5c88dd3f02063L14), [link](https://github.com/JesusFilm/core/pull/1940/files?diff=unified&w=0#diff-3873b898541aff5b78e0bc4aba59192c3d0e767d86c2e57185f5c88dd3f02063L28-R32), [link](https://github.com/JesusFilm/core/pull/1940/files?diff=unified&w=0#diff-3873b898541aff5b78e0bc4aba59192c3d0e767d86c2e57185f5c88dd3f02063L51-R46), [link](https://github.com/JesusFilm/core/pull/1940/files?diff=unified&w=0#diff-fd55f0b59a40cc07f4ea8ad5b314119563477d59a19267ed74705d84595e668aL22-R23), [link](https://github.com/JesusFilm/core/pull/1940/files?diff=unified&w=0#diff-e9e60ec48f7928f205dea4cd73b81cd0498ead851b3bf03af91d9a24b3055ba8L30-R31))
* Replace `JourneyProvider` with `AdminJourneyProvider` and remove unused imports in publisher pages (`apps/journeys-admin/pages/publisher/[journeyId].tsx`, `apps/journeys-admin/pages/publisher/[journeyId]/edit.tsx`) ([link](https://github.com/JesusFilm/core/pull/1940/files?diff=unified&w=0#diff-3ae0c0d27edd7898984c6fa87e97d12383c33f7d0d4558f81384aa19db923d4cL13-R13), [link](https://github.com/JesusFilm/core/pull/1940/files?diff=unified&w=0#diff-3ae0c0d27edd7898984c6fa87e97d12383c33f7d0d4558f81384aa19db923d4cL23-R23), [link](https://github.com/JesusFilm/core/pull/1940/files?diff=unified&w=0#diff-3ae0c0d27edd7898984c6fa87e97d12383c33f7d0d4558f81384aa19db923d4cL48-R38), [link](https://github.com/JesusFilm/core/pull/1940/files?diff=unified&w=0#diff-3ae0c0d27edd7898984c6fa87e97d12383c33f7d0d4558f81384aa19db923d4cL62-R56), [link](https://github.com/JesusFilm/core/pull/1940/files?diff=unified&w=0#diff-3ae0c0d27edd7898984c6fa87e97d12383c33f7d0d4558f81384aa19db923d4cL80-R71), [link](https://github.com/JesusFilm/core/pull/1940/files?diff=unified&w=0#diff-5f65f94b410ccd96a91dea03413e2acb6a41a1d32a1aa49b77395b99952de8acL14), [link](https://github.com/JesusFilm/core/pull/1940/files?diff=unified&w=0#diff-5f65f94b410ccd96a91dea03413e2acb6a41a1d32a1aa49b77395b99952de8acL22-R23), [link](https://github.com/JesusFilm/core/pull/1940/files?diff=unified&w=0#diff-5f65f94b410ccd96a91dea03413e2acb6a41a1d32a1aa49b77395b99952de8acL29-R30), [link](https://github.com/JesusFilm/core/pull/1940/files?diff=unified&w=0#diff-5f65f94b410ccd96a91dea03413e2acb6a41a1d32a1aa49b77395b99952de8acL45-R58), [link](https://github.com/JesusFilm/core/pull/1940/files?diff=unified&w=0#diff-5f65f94b410ccd96a91dea03413e2acb6a41a1d32a1aa49b77395b99952de8acL69-R69))
* Add `tags` field to `journey` values in editor components and tests (`apps/journeys-admin/src/components/Editor/ActionDetails/data.ts`, `apps/journeys-admin/src/components/Editor/ActionsTable/ActionsTable.spec.tsx`, `apps/journeys-admin/src/components/Editor/ControlPanel/Attributes/Action/Action.stories.tsx`, `apps/journeys-admin/src/components/Editor/ControlPanel/Attributes/Action/NavigateToJourneyAction/NavigateToJourneyAction.spec.tsx`, `apps/journeys-admin/src/components/Editor/ControlPanel/Attributes/Action/NavigateToJourneyAction/NavigateToJourneyAction.stories.tsx`, `apps/journeys-admin/src/components/Editor/ControlPanel/Attributes/blocks/Card/BackgroundColor/BackgroundColor.spec.tsx`, `apps/journeys-admin/src/components/Editor/ControlPanel/Attributes/blocks/Card/BackgroundMedia/BackgroundMedia.spec.tsx`, `apps/journeys-admin/src/components/Editor/ControlPanel/Attributes/blocks/Card/BackgroundMedia/BackgroundMedia.stories.tsx`, `apps/journeys-admin/src/components/Editor/ControlPanel/Attributes/blocks/Card/BackgroundMedia/Image/BackgroundMediaImage.spec.tsx`, `apps/journeys-admin/src/components/Editor/ControlPanel/Attributes/blocks/Card/Card.spec.tsx`) ([link](https://github.com/JesusFilm/core/pull/1940/files?diff=unified&w=0#diff-b7b1f79364f0396f28d349443ede37da657501355cc7e0fd320ebb030431f77eL140-R141), [link](https://github.com/JesusFilm/core/pull/1940/files?diff=unified&w=0#diff-a8201af9a19fd03b783c17b766f3f9530c249f468b36b59d910b5fb177b9e4f4L109-R110), [link](https://github.com/JesusFilm/core/pull/1940/files?diff=unified&w=0#diff-d831fc9f24586f8ffc32f5a16edd44fc2d60ddfd43bfddb582c20935cb64fbffL62-R63), [link](https://github.com/JesusFilm/core/pull/1940/files?diff=unified&w=0#diff-7660960fb6dbf2f6f7efc7b474629e215b369a8e0b0805a139e90a391022b073L58-R59), [link](https://github.com/JesusFilm/core/pull/1940/files?diff=unified&w=0#diff-f6da81d812ba404f97df41765dac4693b2f62d3467ed82bf9871ac9a55f79e63L63-R64), [link](https://github.com/JesusFilm/core/pull/1940/files?diff=unified&w=0#diff-b5d409c8300a9bc2d9dc714cbd18cc70a74c1071409335507ee502d4f79dee6bL59-R60), [link](https://github.com/JesusFilm/core/pull/1940/files?diff=unified&w=0#diff-bf130575028e6f024ceb2312a82db951746156cb0e20b838ec4e9359459430ebL60-R61), [link](https://github.com/JesusFilm/core/pull/1940/files?diff=unified&w=0#diff-1197bb2427d836c14e8e79d246692e5ec1648fa5ff88485fce37e32c1ef75f50L75-R76), [link](https://github.com/JesusFilm/core/pull/1940/files?diff=unified&w=0#diff-1a5fff924efb0ab3d388f869c66118b3f4de285acb666e50fec67501f65c5785L70-R71), [link](https://github.com/JesusFilm/core/pull/1940/files?diff=unified&w=0#diff-c0624bf7442dbae5d88b12bba90d9e7b30890825284161b4599d5aed14d3d3c7L95-R96))
* Replace `useJourney` with `useAdminJourney` custom hook in `ActionsTable` component (`apps/journeys-admin/src/components/Editor/ActionsTable/ActionsTable.tsx`) ([link](https://github.com/JesusFilm/core/pull/1940/files?diff=unified&w=0#diff-1e4daaaf44b8fb09b28464e3ae4c952f9fbe8e09737bec92bd5e4e9935077d36L4-R7), [link](https://github.com/JesusFilm/core/pull/1940/files?diff=unified&w=0#diff-1e4daaaf44b8fb09b28464e3ae4c952f9fbe8e09737bec92bd5e4e9935077d36L23-R22))

### CODE REVIEW

File: apps/journeys-admin/src/libs/useAdminJourneyQuery/useAdminJourneyQuery.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+  JourneyStatus,
+  ThemeMode,
+  ThemeName,
+  UserJourneyRole
+} from '../../../__generated__/globalTypes'
+
+import { GET_ADMIN_JOURNEY, useAdminJourneyQuery } from './useAdminJourneyQuery'
+
+describe('useJourneyQuery', () => {
+  it('should get journey', async () => {
```
Comment (csiyang) [inline apps/journeys-admin/src/libs/useAdminJourneyQuery/useAdminJourneyQuery.spec.tsx:19]:
```suggestion
describe('useAdminJourneyQuery', () => {
  it('should get admin journey', async () => {
```

File: apps/journeys-admin/__generated__/GetJourney.ts [hunk 2]
```diff
@@ -625,10 +626,46 @@ export interface GetJourney_journey_host {
   src2: string | null;
 }
 
+export interface GetJourney_journey_team_userTeams_user {
```
Comment (csiyang) [inline apps/journeys-admin/__generated__/GetJourney.ts:629]: @Kneesal can you find whats making these changes? What ever was using this should probably be using the `GetAdminJourney` instead

**Review by csiyang — CHANGES_REQUESTED** [review]

File: apps/journeys-admin/__generated__/GetJourney.ts (hunk 2 above)
Comment (Kneesal) [reply apps/journeys-admin/__generated__/GetJourney.ts:629]: I believe it's coming from useJourneyQuery.ts `apps/journeys-admin/src/libs/useJourneyQuery/useJourneyQuery.ts`

image

Comment (Kneesal) [reply apps/journeys-admin/__generated__/GetJourney.ts:629]: also see: https://github.com/JesusFilm/core/pull/1928/files#diff-9fd061fb3ff27b666fde7550454dc369ec05a5dc7ffef805d83f27b5b40f7b97

I believe this query was made to be used in public pages for journeys-admin, maybe the variable name of the query makes it look like it's only meant to be used in Journeys - however based on the PR above, seems to not be the case

Comment (csiyang) [reply apps/journeys-admin/__generated__/GetJourney.ts:629]: This makes the `useAdminJourney` and `useJourney` the same hook. The query and the return are all the same (except for a name change). 

If we need a Journey without the admin fields we should keep the `useJourney` hook as is. But if we always need all the admin fields, we should just remove the `useJourney` hook.

Comment (Kneesal) [reply apps/journeys-admin/__generated__/GetJourney.ts:629]: yep, due to timeconstraints and large scope, we won't have the time to do that at the moment. Ziwei has made an acknowledgement to this in her previous PR in this comment: 

image

**Comment by Kneesal** [comment]
closed due to refactor of this component being done in this PR
https://github.com/JesusFilm/core/pull/1944

**Pull request closed without merging**